frontend/device/kindle/device.lua

```diff
@@ -168,14 +232,15 @@ local Kindle = Generic:extend{
 }
 
 function Kindle:initNetworkManager(NetworkMgr)
-    function NetworkMgr:turnOnWifi(complete_callback)
+    function NetworkMgr:turnOnWifi(complete_callback, interactive)
         kindleEnableWifi(1)
+        return self:reconnectOrShowNetworkMenu(complete_callback, interactive)
         -- NOTE: As we defer the actual work to lipc,
         --       we have no guarantee the Wi-Fi state will have changed by the time kindleEnableWifi returns,
         --       so, delay the callback until we at least can ensure isConnect is true.
-        if complete_callback then
-            NetworkMgr:scheduleConnectivityCheck(complete_callback)
-        end
+        -- if complete_callback then
+        --     NetworkMgr:scheduleConnectivityCheck(complete_callback)
+        -- end
```

Comment:
I don't recall how this all works, but since we do the same on Kobo, I'm going to assume the connectivity check still fires as expected, so you can just drop the commented out code ;).